.github/workflows/ci.yml

```diff
@@ -39,4 +37,67 @@ jobs:
     - name: Lint Python Code
       run: |
             pip install ruff
-            ruff check
\ No newline at end of file
+            ruff check
+
```

Comment:
Is it possible if we can separate jobs based on the type of application (Job KcMWM, Job TSP, Job KSAT), or do you think it is better to have a job for each check (each run is a job: Job Demo KSAT, Job Python Wrapper KSAT, Job Python Pure KSAT, Job Unit Test KSAT, etc.) so we can easily check the output and debug?

Reply:
Just to clarify, you mean something like a `ksat`-job, `kcmwm`-job, and `tsp`-job, where we run the CPP, pure python, and python wrapper codes for each of those respective jobs?

Reply:
Yes `ksat`-job, `kcmwm`-job, `tsp`-job
OR
something like a `ksat-CPP`-job, `ksat-pure python`-job, `ksat-python-wrapper`-job, `ksat-unit test`-job, `kcmwm-CPP`-job, `kcmwm-pure python`-job, `kcmwm-python-wrapper`-job, `kcmwm-unit test`-job, etc.

Reply:
How do you like the organization of the jobs now? @RustomAlexios

Reply:
I think having `ksat`-job, `kcmwm`-job, `tsp`-job is good, this would include demo, pure python code, python wrapper code. For unit tests, I think it would be good to have three other jobs `ksat-test`-job, `kcmwm-test`-job, `tsp-test`-job so we can easily debug any issues. What do you think?

Reply:
I like the organization of this. I went ahead and updated the pipeline to mirror what you just described